src/soroban.js

```diff
@@ -0,0 +1,72 @@
+import BigNumber from 'bignumber.js';
+
+/**
+ * Soroban helper class
+ * formatting, parsing, and etc
+ * @class Soroban
+ */
+export class Soroban {
+  /**
+   * format token amount to its ideal specific number of decimals
+   *
+   * @param {BigNumber} amount - the token amount you want to display
+   * @param {number} decimals
+   * @returns {string}
+   */
+  static formatTokenAmount(amount, decimals) {
+    let formatted = amount.toString();
+
+    if (decimals > 0) {
+      formatted = amount.shiftedBy(-decimals).toFixed(decimals).toString();
+
+      // Trim trailing zeros
+      while (formatted[formatted.length - 1] === '0') {
+        formatted = formatted.substring(0, formatted.length - 1);
+      }
+
+      if (formatted.endsWith('.')) {
+        formatted = formatted.substring(0, formatted.length - 1);
+      }
+    }
+
+    return formatted;
+  }
+
+  /**
+   * parse token amount to use it on smart contract
```

Comment:
oh derp the return value is a whole number not a fraction, so it would return `10012300`, right?

Reply:
yup yup

Reply:
hmm, I think that means we can do `bigint(whole) * decimal` + `bigint(frac_padded_with_decimal_zeroes)`?? :thinking:

Reply:
oh do you mean to do that instead of this?
`wholeValue.shiftedBy(decimals).plus(fractionValue);`

Hmm does `bigint(whole) * decimal` == `wholeValue.shiftedBy(decimals).`? I think it would be something like `bigint(whole) * (decimal * 10)` in order to move the decimal by the number of places defined by the decimal?

Reply:
oh yeah that's what I meant hehe nice catch - I'm doing off-the-cuff napkin math

Reply:
👍 I think that checks out

Reply:
wait LOL isn't this `whole + part.padEnd(decimals, '0')`??? I just tried it:

```javascript
// inputs 
let value = "123.4560";
let decimals = 5;

// algo
let [ whole, part ] = value.split('.').slice(0, 2);
let shifted = BigInt(whole + part.padEnd(decimals, '0'));
console.log(shifted.toString())

// "12345600"
```

Reply:
hmm yeah I think that does work, but still squinting at that nullish coalesce.
It looks to me like `shifted` could == `undefined00` for the whole number case because the coalesce is happening on `part?.padEnd(decimals, '0')` and not just on `part`, right?

Reply:
but `part?.padEnd` does not call `padEnd` if `part === null | undefined`, so it hits the right side of the `??` case, right?

Reply:
ah yup, I see. Thanks!